app/forms/hub/create_ctc_client_form.rb

```diff
@@ -148,5 +158,45 @@ def tax_return_attributes
         service_type: :drop_off
       }.merge(attributes_for(:tax_return))
     end
+
+    def selected_drivers_license?
+      with_drivers_license_photo_id == "1"
+    end
+
+    def selected_passport?
+      with_passport_photo_id == "1"
+    end
+
+    def selected_other_state_id?
+      with_other_state_photo_id == "1"
+    end
+
+    def selected_vita_approved_photo_id?
+      with_vita_approved_photo_id == "1"
+    end
+
+    def selected_vita_approved_taxpayer_id?
+      with_vita_approved_taxpayer_id == "1"
+    end
+
+    def selected_social_security?
+      with_social_security_taxpayer_id == "1"
+    end
+
+    def selected_itin?
+      with_itin_taxpayer_id == "1"
+    end
+
+    def at_least_one_photo_id_type_selected
```

Comment:
Can we leverage PHOTO_ID_TYPES and TAXPAYER_ID_TYPES to create this list of acceptable photo id types? Instead of different predicate methods for each id type, you could create one method that employs :send like your methods on Intake.

```
def selected_photo_id_type
  has_photo_id = false
  while has_photo_id == false
    Intake::PHOTO_ID_TYPES.each do |type|
       has_photo_id = self.send(type[:field_name] == 1
    end
  end
  errors.add(...) unless has_photo_id
end
```

Reply:
Good call, will do!